streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -907,6 +905,17 @@ private void populatePartitionsByHostMaps(final Map<HostInfo, Set<TopicPartition
                 minSupportedMetadataVersion,
                 false,
                 encodeNextProbingRebalanceTime);
+
+            if (followupRebalanceScheduled) {
+                rebalanceRequired = true;
+                log.debug("Requested client {} to schedule a followup rebalance", clientId);
+            }
+        }
+
+        if (rebalanceRequired) {
```

Comment:
Previously we were doing this logging at the client-level, meaning we could end up in the confusing situation of logging `Stable rebalance..` for one client and `Unstable rebalance...` for another, all in the same rebalance. Now we only log the final result after all client assignments are encoded, and log which clients actually are scheduling said rebalance at the debug level.